app/parse.py

```diff
@@ -12,7 +18,71 @@ class Course:
     name: str
     short_description: str
     course_type: CourseType
+    modules_count: int
+    topics_count: int
+    duration_in_months: int
+
+
+def parse_additional_info(course_link: str) -> tuple:
+    page = requests.get(urljoin(BASE_URL, f"{course_link}")).content
+    soup = BeautifulSoup(page, "html.parser")
+
+    modules, topics, duration = [
+        int(additional_info.text.split()[0])
+        for additional_info in soup.select(
+            "div.CourseModulesHeading_headingGrid__50qAP p"
+        )
+    ]
+    return modules, topics, duration
+
+
+def parse_course(course_soup: Tag) -> list[Course]:
+    courses = []
+
+    course_link_list = [
+        link["href"] for link in course_soup.select("[data-qa]")
+    ]
+
+    course_type_list = [
+        course_type.text for course_type in course_soup.select("a span")
+    ]
+
+    for course_type, course_link in zip(course_type_list, course_link_list):
+
+        modules, topics, duration = parse_additional_info(course_link)
+
+        if course_type == "Власний темп":
+            course_type = CourseType.PART_TIME
+
+        if course_type == "Повний день":
+            course_type = CourseType.FULL_TIME
```

Comment:
I think better to create `CourseParser` for example
And move all this functions to class